lib/safira/interaction/spotlight.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Safira.Interaction.Spotlight do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "spotlights" do
+    field :active, :boolean, default: false
+    field :badge_id, :id
+    field :lock_version, :integer, default: 1
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(spotlight, attrs) do
+    spotlight
+    |> cast(attrs, [:active, :badge_id])
+    |> validate_required([:active, :badge_id])
+    |> validate_not_already_active()
+    |> Ecto.Changeset.optimistic_lock(:lock_version)
+  end
+
+  def finish_changeset(spotlight, attrs) do
+    spotlight
+    |> cast(attrs, [:active])
+    |> validate_required([:active])
+  end
+
+  defp validate_not_already_active(changeset) do
+    if changeset.data && changeset.data.active do
+      add_error(changeset, :active, "Another spotlight is still active")
+    else
+      changeset
+    end
```

Comment:
@joaop21 I used otimistic locking as you advised, please take a look at this snippet of code and the Multi used in the Interaction context and tell me if the correct concurrency control has been achieved.

Reply:
If that changeset.data includes the spotlight, I guess it's fine.

Reply:
Yes, because a spotlight will always be created previously, so this changeset is only used for updating the existing spotlight, ensuring the `data` field always exists.